pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,153 +1,226 @@
+  hasFreetype = lib.elem fontBackend [
+    "coretext_freetype"
+    "fontconfig_freetype"
+    "freetype"
+  ];
+
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/font/main.zig#L144
+  hasHarfbuzz = lib.elem fontBackend [
+    "coretext_freetype"
+    "coretext_harfbuzz"
+    "fontconfig_freetype"
+    "freetype"
+  ];
 in
+
 stdenv.mkDerivation (finalAttrs: {
   pname = "ghostty";
   version = "1.0.0";
 
+  outputs = [
+    "out"
+    "man"
+    "terminfo"
+  ];
+
   src = fetchFromGitHub {
     owner = "ghostty-org";
     repo = "ghostty";
     tag = "v${finalAttrs.version}";
     hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
   };
 
-  strictDeps = true;
-
-  nativeBuildInputs = [
-    glib # Required for `glib-compile-schemas`
-    ncurses
-    pandoc
-    pkg-config
-    removeReferencesTo
-    wrapGAppsHook4
-    zig_hook
-  ];
-
-  buildInputs = [
-    bzip2
-    expat
-    fontconfig
-    freetype
-    glslang
-    harfbuzz
-    libadwaita
-    libGL
-    libpng
-    libX11
-    libXcursor
-    libXi
-    libXrandr
-    oniguruma
-    zlib
-  ];
-
-  dontConfigure = true;
-  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
-  doCheck = false;
-  doInstallCheck = true;
+  # Avoid using runtime hacks to help find X11
+  postPatch = lib.optionalString (appRuntime == "gtk") ''
+    substituteInPlace src/apprt/gtk/x11.zig \
+      --replace-warn 'std.DynLib.open("libX11.so");' 'std.DynLib.open("${lib.getLib libX11}/lib/libX11.so");'
+  '';
 
   deps = callPackage ./deps.nix {
     name = "${finalAttrs.pname}-cache-${finalAttrs.version}";
   };
 
+  strictDeps = true;
+
+  nativeBuildInputs =
+    [
+      ncurses
+      pandoc
+      pkg-config
+      removeReferencesTo
+      zig_hook
+    ]
+    ++ lib.optionals (appRuntime == "gtk") [
+      glib # Required for `glib-compile-schemas`
+      wrapGAppsHook4
+    ];
+
+  buildInputs =
+    [
+      glslang
+      oniguruma
+    ]
+    ++ lib.optional (appRuntime == "gtk" && withAdwaita) libadwaita
+    ++ lib.optional (appRuntime == "gtk") libX11
+    ++ lib.optional (renderer == "opengl") libGL
+    ++ lib.optional hasFontconfig fontconfig
+    ++ lib.optional hasHarfbuzz harfbuzz
+    ++ lib.optionals hasFreetype [
+      bzip2
+      freetype
+    ];
+
   zigBuildFlags =
     [
       "--system"
       "${finalAttrs.deps}"
       "-Dversion-string=${finalAttrs.version}"
+
+      "-Dapp-runtime=${appRuntime}"
+      "-Dfont-backend=${fontBackend}"
+      "-Dgtk-adwaita=${lib.boolToString withAdwaita}"
+      "-Drenderer=${renderer}"
     ]
     ++ lib.mapAttrsToList (name: package: "-fsys=${name} --search-prefix ${lib.getLib package}") {
       inherit glslang;
     };
 
   zigCheckFlags = finalAttrs.zigBuildFlags;
 
-  outputs = [
-    "out"
-    "terminfo"
-    "shell_integration"
-    "vim"
```

Comment:
This supports `programs.vim.plugins = [ ghostty.vim ]`. What would you suggest is the alternative? (neovim is just fine with `programs.vim.plugins = [ ghostty ]`)

Reply:
I would say a separate package that handles the vim integration would be better in my opinion as ghostty can be installed on headless systems, then if you need the vim support then you install the vim support separately if you need the vim support.

Reply:
You'll get the `/share/vim/vim-files/*` output from the build regardless of whether you want them or not. (along with several other integrations that work off of `$out/share`). This is much simpler to maintain than a parallel derivation. (or a derivation that exists only to move the set of folders under `vim-files` to the root of the derivation)

Have a look inside $out/share one your system, all kinds of extra integrations 'just work' in nixos because the build puts things there.

Reply:
getchoo's choice not to add it to outputs to install is just fine.